### PR TITLE
Feature/camera

### PR DESCRIPTION
**Pull request by Sonusiety**

### CODE REVIEW

File: app/src/main/java/com/github/sdpsharelook/camera/CameraActivity.kt [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+        ContextCompat.checkSelfPermission(
+            this,
+            permission
+        ) == PackageManager.PERMISSION_GRANTED -> {
+
+        }
+        else -> {
+            requestPermissionLauncher.launch(
+                permission
+            )
```
Comment (rombirli) [inline app/src/main/java/com/github/sdpsharelook/camera/CameraActivity.kt:95]: I did a similar code for permission when doing the speechRecognizer. We should consider extract this part in a dedicated class.

File: app/src/main/java/com/github/sdpsharelook/camera/CameraActivity.kt [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+        checkPermissions(Manifest.permission.CAMERA)
+        if (checkPerms()) {
+            cameraLauncher.launch(tempImageUri)
+        }
+    }
+
+    private var tempImageUri: Uri? = null
+    private val cameraLauncher = registerForActivityResult(ActivityResultContracts.TakePicture()) { success ->
+        if (success) {
+            val file = File(currentPath)
```
Comment (rombirli) [inline app/src/main/java/com/github/sdpsharelook/camera/CameraActivity.kt:62]: Maybe we should clearly define later what goes to the main storage and what feeds the cache. I think not all pictures should be saved for lifetime.

**Review by rombirli — APPROVED** [review]
Good work overall !

File: app/src/main/res/layout/activity_camera.xml [hunk 3]
```diff
@@ -0,0 +1,44 @@
+<?xml version="1.0" encoding="utf-8"?>
+<androidx.constraintlayout.widget.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    tools:context=".camera.CameraActivity">
+
+    <!--    <ImageView-->
```
Comment (villemontc) [inline app/src/main/res/layout/activity_camera.xml:9]: Remove those comments

**Review by villemontc — APPROVED** [review]
Great work but clean your code before merge

**Review by AdrianUlrich — APPROVED** [review]
It works, but would benefit from some code cleanup

**Pull request merged**